setup/tue-functions.bash

```diff
@@ -572,6 +572,129 @@ function _show_file
     fi
 }
 
+function __generate_setup_file
+{
+    # Check whether this target was already added to the setup
+    if [[ "$TUE_SETUP_TARGETS" == *" $1 "* ]];
+    then
+        return 0
+    fi
+
+    TUE_SETUP_TARGETS=" $1$TUE_SETUP_TARGETS"
+
+    # Check if the dependency file exists. If not, return
+    if [ ! -f "$tue_dependencies_dir"/"$1" ]
+    then
+        return 0
+    fi
+
+    # Recursively add a setup for each dependency
+    deps=$(cat "$tue_dependencies_dir"/"$1")
+    for dep in $deps
+    do
+        # You shouldn't depend on yourself
+        if [ "$1" != "$dep" ]
+        then
+            __generate_setup_file "$dep"
+        fi
+    done
+
+    local tue_setup_file=$TUE_ENV_TARGETS_DIR/$1/setup
+    if [ -f "$tue_setup_file" ]
+    then
+        echo "source $tue_setup_file" >> "$TUE_ENV_DIR"/.env/setup/target_setup.bash
+    fi
+}
+
+function _generate_setup_file
+{
+    mkdir -p "$TUE_ENV_DIR"/.env/setup
+    echo "# This file was auto-generated by tue-get. Do not change this file." > "$TUE_ENV_DIR"/.env/setup/target_setup.bash
+
+    tue_dependencies_dir="$TUE_ENV_DIR"/.env/dependencies
+
+    mkdir -p "$tue_dependencies_dir"
+    local installed_targets
+    installed_targets=$(ls "$tue_dependencies_dir")
+    TUE_SETUP_TARGETS=" "
+    for t in $installed_targets
+    do
+        __generate_setup_file "$t"
```

Comment:
The function can be removed as it is only called here. Instead condense the code and put it in the for loop. The logic becomes more readable.

Reply:
Also this would eliminate the need for passing a variable as a global variable to the function.

Reply:
This function is called recursive...........

Reply:
Are local variables available in the functions called?

Reply:
Put it in the for loop? Which?

Reply:
Both `__generate_setup_file ` and `_generate_setup_file ` are needed. `__generate_setup_file ` is called recursively. `_generate_setup_file ` is called on two spots